src/core/smallEvents/bonusGuildPVEIslandSmallEvent.ts

```diff
@@ -88,38 +123,7 @@ export const smallEvent: SmallEvent = {
 		const event = tr.getRandomFromKeys("events");
 		const intro = tr.get(`events.${event}.intro`);
 
-		let sentence;
-		if (player.isInGuild()) {
-			const nbMemberRequired = RandomUtils.randInt(1, 4);
-			const playersOnPVEIsland = await Maps.getGuildMembersOnPveIsland(player);
-			if (playersOnPVEIsland.length >= nbMemberRequired) {
-				sentence = tr.format(`events.${event}.success.withGuild`, {
-					amount: await applyPossibility({interaction, language, tr}, player, `${event}.success.withGuild.malus`)
-				});
-				seEmbed.setDescription(`${seEmbed.data.description}${intro}\n\n${sentence}`);
-				await interaction.editReply({embeds: [seEmbed]});
-				return;
-			}
-		}
-
-		const probabilities = RandomUtils.randInt(0, 100);
-		if (probabilities < 10) {
-			sentence = player.isInGuild() ? tr.format(`events.${event}.success.soloWithGuild`, {
-				amount: await applyPossibility({interaction, language, tr}, player, `${event}.success.withGuild.malus`)
-			}) : tr.format(`events.${event}.success.solo`,{
-				amount: await applyPossibility({interaction, language, tr}, player, `${event}.success.solo.malus`)
-			});
-		}
-		else if (probabilities < 40) {
-			sentence = player.isInGuild() ? tr.get(`events.${event}.escape.withGuild`) : tr.get(`events.${event}.escape.solo`);
-		}
-		else {
-			sentence = player.isInGuild() ? sentence = tr.format(`events.${event}.lose.withGuild`, {
-				amount: await applyPossibility({interaction, language, tr}, player, `${event}.lose.withGuild.malus`)
-			}) : tr.format(`events.${event}.lose.solo`, {
-				amount: await applyPossibility({interaction, language, tr}, player, `${event}.lose.solo.malus`)
-			});
-		}
+		const sentence = await drawPossibilities({interaction, language, tr}, player, event, player.isInGuild() ? await hasEnoughMemberOnPVEIsland(player) : false);
```

Comment:
pour moi le player is in guild doit faire partie de la fonction qui check si ya assez de pote avec lui